modules/trusted-firmware-m/Kconfig

```diff
@@ -31,19 +31,22 @@ config TFM_MINIMAL
 
 if !TFM_MINIMAL
 rsource "Kconfig.mbedtls.defconfig"
-source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig"
+
+# Here we need to duplicate everything from
+# '$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig' except 'Kconfig.tfm'.
+# This because 'Kconfig.tfm' will be sourced regardless of the value of
+# 'TFM_MINIMAL'
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.partitions"
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules"
 endif
 
 if TFM_MINIMAL
-
 rsource "Kconfig.mbedtls_minimal.defconfig"
-
 rsource "Kconfig.tfm_minimal.defconfig"
+endif # TFM_MINIMAL
 
 source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"
```

Comment:
Should we move this line above line 32, that is:
```
source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"

if !TFM_MINIMAL
rsource "Kconfig.mbedtls.defconfig"

```

That wil serve two purposes:
- Same sourcing order:
  - Kconfig.tfm
  - Kconfig.tfm.partitions
  - Kconfig.tfm.crypto_modules
  as in upstream is guaranteed. https://github.com/zephyrproject-rtos/zephyr/blob/00d707e470e66309f08fbf94b42580c9c272d969/modules/trusted-firmware-m/Kconfig#L7-L10
- Sourcing `Kconfig.tfm` right before `if !TFM_MINIMAL` makes is more visible that it is always sourced, and hence the comment doesn't have two be that long, but could simply be something like:
```
source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"

if !TFM_MINIMAL
rsource "Kconfig.mbedtls.defconfig"

# When enabling full TFM support, then partitions and crypto modules shall be available.
source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.partitions"
source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules"
endif

```